srcStatic/ConsoleModuleLoader.cpp

```diff
@@ -12,8 +12,19 @@
 #include <system_error>
 
 
-ConsoleModuleLoader::ConsoleModuleLoader(const std::string& moduleRelativeDirectory)
+// Console module names are the relative path from the game folder (no trailing slash)
+ConsoleModuleLoader::ConsoleModuleLoader(const std::vector<std::string>& moduleNames)
 {
+	if (moduleNames.empty()) {
+		return; // No console modules to load
+	}
+
+	// For now just handle the first name
+	if (moduleNames.size() > 1) {
+		throw std::runtime_error("ConsoleModuleLoader currently only supports a single loaded module");
```

Comment:
This exception would be unhandled if it were thrown. It would be thrown during DLL initialization, so not really a way to catch it.

It should never be thrown because line 21 of ConsoleArgumentParser will abort when it detects multiple modules being loaded.

```C++
PostErrorMessage(__FILE__, __LINE__, "Error parsing command line arguments: " + std::string(e.what()));
```

This makes it seem a bit sloppy though. 

I understand this leaves the code in an intermediate state so I don't want to spend lots of time finding the perfect answer. If this used the PostErrorMessage and then returned without loading a module, it would keep from throwing the unhandled exception. @DanRStevens, what do you think?

Reply:
Well, yes, it is a bit sloppy. Though that is kind of by design. There is of course no intention to leave the code in that state. The check was added because I had chosen to defer work on the internals of the class until a later branch. The check is a temporary reminder that the guts of the class haven't yet been updated to support multiple modules. Only the interface has been updated so it could potentially support multiple modules.

More to the immediate point of the refactoring done in this branch, the constructor will only ever be passed a vector of 1 element. The caller is hardcoded to construct a vector of 1 element. That means the check will never be triggered in a production environment.

Ultimately the limitation on the constructor call does come from the `ConsoleArgumentParser` design. The function signature is set to return only a single value. Hence when that value is passed to the `ConsoleModuleLoader`, it is wrapped into a vector of 1 element, just to satisfy the interface change.

The plan is to update the internals before updating the command line argument processing.

If for some reason we do start passing multiple modules before the internals support it, that is a pretty serious logic flaw, and you might argue the program should crash (in a controlled manner).

Or maybe I'm just going about things backwards by updating the interface before updating the internals.

Reply:
I can accept that this will be quickly modified again as an intermediate step.

I suppose terminating the program with an unhandled exception might be okay here (although I don't think I would prefer it), but it would still at least need to be logged why the program was terminated.